### ISSUE TITLE
Tensorflow CudnnGRU

### ISSUE THREAD

**Issue by ilkarman** [ilkarman/DeepLearningFrameworks#29]
I'm trying to replace the basic GRU Cell I [currently have](https://github.com/ilkarman/DeepLearningFrameworks/blob/master/Tensorflow_RNN.ipynb):

```
cell = tf.contrib.rnn.GRUCell(NUMHIDDEN)
outputs, states = tf.contrib.rnn.static_rnn(cell, word_list, dtype=tf.float32)

```
With the CuDNN version:

```
cudnn_cell = tf.contrib.cudnn_rnn.CudnnGRU(num_layers=1, 
                                           num_units=NUMHIDDEN, 
                                           input_size=EMBEDSIZE)    # Set params
params_size_t = cudnn_cell.params_size()
params = tf.Variable(tf.ones([params_size_t]), validate_shape=False)   
input_h = tf.Variable(tf.ones([1, BATCHSIZE, NUMHIDDEN]))

outputs, states = cudnn_cell(is_training=True,
                             input_data=word_list,
                             input_h=input_h,
                             params=params)
```

However, when I do this my model starts to predict randomly. My accuracy goes from 0.86 to 0.5

**Comment by traveller59** [comment]
use tf.random_uniform instead of tf.ones
```Python
params = tf.Variable(tf.random_uniform([params_size_t], -0.1, 0.1), validate_shape=False)
```

**Comment by ilkarman** [comment]
Thanks @traveller59 for the suggestion, it seems I was initialising in a crappy way. It's [better now ](https://github.com/ilkarman/DeepLearningFrameworks/blob/master/Tensorflow_RNN.ipynb) but stuck at 77% instead of 83 (like the others). I'm not sure if it's because it wants an is_training flag but since I don't have any recurrent dropout I don't see why this would make a difference?

**Comment by ilkarman** [comment]
Thanks @traveller59 for the suggestion, that helped! It's [better now ](https://github.com/ilkarman/DeepLearningFrameworks/blob/master/Tensorflow_RNN.ipynb) but stuck at 77% instead of 83 (like the others). I'm not sure if it's because it wants an is_training flag but since I don't have any recurrent dropout I don't see why this would make a difference? If it wants an is_training flag, since it doesn't support tf.placeholder I can't use that with feeddict, would I then have to create two different graphs for training/inference?

**Comment by traveller59** [comment]
the default range of tf.random_uniform is [0, 1], it‘s still too large for RNN, [-0.1, 0.1] is recommended. I have run your code with range [-0.1, 0.1] and test accuracy can reach 0.85

**Comment by ilkarman** [comment]
Thank you! That was very helpful!
